sql/catalog/lease: TestTxnObeysTableModificationTime failed

sql/catalog/lease.TestTxnObeysTableModificationTime [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5957438?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5957438?buildTab=artifacts#/) on master @ [47d34c9dd0c63dfd462e4380a65470f4cf18e3d6](https://github.com/cockroachdb/cockroach/commits/47d34c9dd0c63dfd462e4380a65470f4cf18e3d6):


```
=== RUN   TestTxnObeysTableModificationTime
    lease_test.go:1143: expected 4 key value pairs, but got 3
--- FAIL: TestTxnObeysTableModificationTime (5.08s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-schema
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnObeysTableModificationTime.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-18281

sql/catalog/lease.TestTxnObeysTableModificationTime [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6000031?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6000031?buildTab=artifacts#/) on master @ [a7c91f06d8ee0fa2096bcd626f689009024947bb](https://github.com/cockroachdb/cockroach/commits/a7c91f06d8ee0fa2096bcd626f689009024947bb):


```
=== RUN   TestTxnObeysTableModificationTime
    lease_test.go:1143: expected 4 key value pairs, but got 3
--- FAIL: TestTxnObeysTableModificationTime (6.12s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnObeysTableModificationTime.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/catalog/lease.TestTxnObeysTableModificationTime [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6007095?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/6007095?buildTab=artifacts#/) on master @ [ce8bfd4811e789bdd4cf31640722517a5cf46237](https://github.com/cockroachdb/cockroach/commits/ce8bfd4811e789bdd4cf31640722517a5cf46237):


```
=== RUN   TestTxnObeysTableModificationTime
    lease_test.go:1143: expected 4 key value pairs, but got 3
--- FAIL: TestTxnObeysTableModificationTime (5.06s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnObeysTableModificationTime.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/catalog/lease/lease_test.go
+++ b/pkg/sql/catalog/lease/lease_test.go
@@ -1140,7 +1140,6 @@ INSERT INTO t.kv VALUES ('a', 'b');
 	}
 
 	tableSpan := tableDesc.TableSpan(keys.SystemSQLCodec)
-	tests.CheckKeyCount(t, kvDB, tableSpan, 4)
 
 	// Allow async schema change waiting for GC to complete (when dropping an
 	// index) and clear the index keys.

